src/server/sample_data/routes/uninstall.js

```diff
@@ -43,12 +43,16 @@ export const createUninstallRoute = () => ({
       }
 
       const { callWithRequest } = server.plugins.elasticsearch.getCluster('data');
-      const index = createIndexName(server, sampleDataset.id);
 
-      try {
-        await callWithRequest(request, 'indices.delete', { index: index });
-      } catch (err) {
-        return reply(`Unable to delete sample data index "${index}", error: ${err.message}`).code(err.status);
+      for (let i = 0; i < sampleDataset.dataIndices.length; i++) {
+        const dataIndexConfig = sampleDataset.dataIndices[i];
```

Comment:
Similar logic is in the `install.js` file. It uninstalls the dataset prior to installing it. Maybe `install.js` should just call this somehow instead of duplicating this?

Reply:
`uninstall` handles deletes differently. If a delete fails, then it replies to the request with the status code of the failure.

`install` just tries a delete to ensure the index is in a clean state before setting mappings.

The error handling is different enough that you would not gain anything by combining these.